lib/kino/bridge.ex

```diff
@@ -250,6 +250,15 @@ defmodule Kino.Bridge do
     match?({:ok, _}, io_request(:livebook_get_evaluation_file))
   end
 
+  @doc """
+  Requests the child spec for proxy handler with the given function.
+  """
+  @spec get_proxy_handler_child_spec((Plug.Conn.t() -> Plug.Conn.t())) ::
+          Supervisor.child_spec() | {module(), term()} | module()
+  def get_proxy_handler_child_spec(fun) do
+    with {:ok, reply} <- io_request({:livebook_get_proxy_handler_child_spec, fun}), do: reply
```

Comment:
```suggestion
    io_request({:livebook_get_proxy_handler_child_spec, fun})
```

and we can adjust the typespec to be `{:ok, {module(), term()}} | request_error()`. Then in `listen` we can match and raise an error accordingly, similarly to

https://github.com/livebook-dev/kino/blob/df73bed09b3363c40e35a47c8272c8d1e7642f2f/lib/kino/hub.ex#L33-L34